src/peer_select.cc

```diff
@@ -925,16 +944,27 @@ peerHandlePingReply(CachePeer * p, peer_t type, AnyP::ProtocolType proto, void *
 }
 
 static void
-peerAddFwdServer(FwdServer ** FSVR, CachePeer * p, hier_code code)
+peerAddFwdServer(ps_state *ps, CachePeer *peer, const hier_code code)
```

Comment:
please take the opportunity to make this function a method of ps_state

Reply:
I actually started doing that before posting this pr but aborted the attempt because I realized that the new method would have to be _public_ and exposing such a low-level operation to all `ps_state` users may not be such a good idea. Let's convert all these static functions with `ps_state` argument into (protected) methods in a single sweep, in a dedicated "no functionality changes" pr.